backend/src/main/java/com/hackner/musicompass/service/RecommendationService.java

```diff
@@ -48,4 +51,53 @@ public RecommendationTags changeRecommendationTags (RecommendationTagsDto recomm
         return recommendationCategoryMongoDb.findAll();
     }
 
+    public String getArtistRecommendation(RecommendationRequestDto recommendationRequestDto){
+
+        List<Artist> artists = artistMongoDb.findAll();
+
+        List<Artist> genreFilteredArtists = filterByRecommendationTags(artists, recommendationRequestDto.getGenres(), "genres");
+        List<Artist> genreRoleFilteredArtists = filterByRecommendationTags(genreFilteredArtists, recommendationRequestDto.getRoles(), "roles");
+        List<Artist> fullyFilteredArtists = filterByRecommendationTags(genreRoleFilteredArtists, recommendationRequestDto.getGender(), "gender");
```

Comment:
I would prefer a chained stream without allocating 3 new ArrayLists

Reply:
I see your point. I didnt manage to do this, because I need an if condition to check if there are any tags or if the tags list is empty...

Reply:
Hm.., I think `.containsall(anEmptyList)` will be always true, but maybe I am mistaken?